utils.go

```diff
@@ -755,21 +740,16 @@ func mntSrcUidShiftNeeded(mntSrc string, uid, gid uint32) (bool, uint32, uint32,
 		return false, 0, 0, err
 	}
 
-	needChown := true
+	numNeedChown := 0
 	for _, fi := range dirFis {
 		st, _ := fi.Sys().(*syscall.Stat_t)
-
-		if uid <= st.Uid && gid <= st.Gid {
-
-			logrus.Infof("skipping uid shift on %s because subdir %s has uid:gid (%d:%d) which is "+
-				">= the container's assigned uid:gid (%d:%d)",
-				mntSrc, fi.Name(), st.Uid, st.Gid, uid, gid)
-
-			needChown = false
-			break
+		if uid != st.Uid && gid != st.Gid {
```

Comment:
Isn't this an OR op?

Reply:
Good point, I changed it to OR. In practice it won't make a difference, because the chown happens on host dirs that get mounted into the container's special dirs (e.g., /var/lib/docker) and for those we always expect the uid:gid to match. But it's clearer / more correct to do the OR instead of AND.